packages/db/src/query/live/collection-subscriber.ts

```diff
@@ -201,10 +198,22 @@ export class CollectionSubscriber<
     }
 
     const trackedChanges = this.trackSentValues(changes, orderByInfo.comparator)
-    this.sendChangesToPipeline(
-      trackedChanges,
-      this.loadMoreIfNeeded.bind(this, subscription)
-    )
+
+    // Cache the bound loader on the subscription using a symbol property.
+    // This ensures we pass the same function instance to the scheduler each time,
+    // allowing it to deduplicate callbacks when multiple changes arrive during a transaction.
+    type SubscriptionWithLoader = CollectionSubscription & {
+      [loadMoreCallbackSymbol]?: () => boolean
+    }
+
+    const subscriptionWithLoader = subscription as SubscriptionWithLoader
+
+    const boundLoader =
+      subscriptionWithLoader[loadMoreCallbackSymbol] ??
```

Comment:
If i'm not mistaked, the trick we do with ?? here is exactly what `??=` does so we can rewrite this as:
```ts
subscriptionWithLoader[loadMoreCallbackSymbol] ??=
  this.loadMoreIfNeeded.bind(this, subscription)
```